Dashboard/Controllers/BuildsController.cs

```diff
@@ -101,6 +101,105 @@ public ActionResult View(bool pr = false, DateTimeOffset? startDate = null, stri
             return View(viewName: "View", model: model);
         }
 
+        /// <summary>
+        /// A view of the elapsed time grouped by the result.
+        /// </summary>
+        /// <returns></returns>
+        public ActionResult ElapsedTime(bool pr = false, DateTimeOffset? startDate = null, string viewName = AzureUtil.ViewNameRoslyn)
+        {
+            var filter = CreateBuildFilter(actionName: nameof(ElapsedTime), viewName: viewName, startDate: startDate, pr: pr);
+            var results =
+                _buildUtil.GetBuildResults(filter.StartDate, viewName)
+                .Where(x => pr || !JobUtil.IsPullRequestJobName(x.JobId))
+                .ToList();
+
+            var totalCount = results.Count;
+            var totalSucceeded = results.Count(x => x.ClassificationKind == ClassificationKind.Succeeded);
+
+            var runCounts = results
+                .Select(x => new ElapsedTimeModel() { JobId = x.JobId, JobName = x.JobName, ElapsedTime = x.DurationSeconds })
+                .ToList();
+
+            List<int> runsPerETRange = new List<int>();
+
+            for (int i = 0; i < 6; i++)
```

Comment:
This array is to draw the 6 rows of ET range on chart http://jdash-et.azurewebsites.net/builds/elapsedtime. As 6 ranges can already cover all jobs (jobs longer then that will time out), and this chart isn't top pri to managers (they care more about the "ElapsedTime" main menu view), I choose fixed number of list instead of a dynamic List here.

Agree the magic number 6 is confusing, even though I have comments in ElapsedTimeSummaryModel declaration of BuildsModel.cs. I've defined a private const int _ETRangeCount = 6 and used that var instead of the magic number 6.

        private const int _ETRangeCount = 6;
...
        for (int i = 0; i < _ETRangeCount; i++)